### PR TITLE
Feature/96 implement visual novel search

### PR DESCRIPTION
**Pull request by NightOfNights**
1) Search feature is not ready yet, it's just a checkpoint;
2) I will remove the form from the visual novel page on merge to develop branch.

### CODE REVIEW

File: src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    control,
+  } = useForm({ defaultValues: visualNovelFormInitialValues });
+
+  /**
+   * Handles submission of visual novel form.
+   * @param data Form data.
+   */
+  function handleVisualNovelFormSubmit(data: VisualNovelFormData): void {
+    /** TODO (Panov A.): Remove console.log. */
+    console.log(data);
```
Comment (Moltenship) [inline src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx]: Do not forget to remove it :)

**Review by Moltenship — APPROVED** [review]

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -8,19 +8,19 @@ interface Props {
   /** Whether logo and sidebar toggle button are visible. */
   isLogoVisible: boolean;
 
-  /** Show sidebar. */
-  onSiderbarShow: () => void;
+  /** Callback to run when the sidebar should show up. */
+  onSidebarShow: () => void;
```
Comment (moetazaneta) [inline src/components/Header/Header.tsx]: Please add `readonly`, in place of some forgetful developer...

File: src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+const platformOptions: SelectOption[] = Platform.getSortedPlatforms().map(
+  platform => ({ value: platform, label: Platform.toReadable(platform) }),
+);
+
+const currentYear = new Date().getFullYear();
+
+const visualNovelFormInitialValues: VisualNovelFormData = {
+  title: '',
+  lengthRange: [0, 50],
+  releaseYearRange: [1970, currentYear],
```
Comment (moetazaneta) [inline src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx]: Consider moving to constants.

File: src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        </FormControl>
+
+        <HStack width="full">
+          <MultiSelect
+            control={control}
+            name="languages"
+            label="Language"
+            options={languageOptions}
+            placeholder="Any"
+            displayLimit={1}
```
Comment (moetazaneta) [inline src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx]: Not sure if it should be moved into constants

File: src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            label="Length"
+            min={0}
+            max={50}
+          />
+          <RangeSlider
+            control={control}
+            name="releaseYearRange"
+            label="Release date"
+            min={1970}
+            max={currentYear}
```
Comment (moetazaneta) [inline src/features/VisualNovel/components/VisualNovelSearchForm/VisualNovelSearchForm.tsx]: That definitely should be moved to constants, as I see it second time.

File: src/utils/formControl.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+ * Props that are required for any form control component.
+ */
+export interface FormControlProps {
+
+  /**
+   * The object from the invoking `useForm` hook of React Hook Forms.
+   * Provides API for registering component into needed form.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  readonly control: Control<any>;
```
Comment (moetazaneta) [inline src/utils/formControl.ts:13]: Can't it be used with `unknown`? not sure

**Review by Moltenship — APPROVED** [review]

File: src/utils/formControl.ts (hunk 6 above)
Comment (NightOfNights) [reply src/utils/formControl.ts:13]: No :(

**Pull request merged**